Structure_Engine/Compute/PanelToFEMesh.cs

```diff
@@ -0,0 +1,105 @@
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.Engine.Analytical;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Converting panel with 3 or 4 control points to FEmesh ")]
+        [Input("panel", "Panel to be converted to an FEMesh.")]
+        [Output("FEMesh", "FEMesh converted from a Panel.")]
+
+        public static FEMesh PanelToFEMesh(Panel panel)
+        {
+            List<Point> points = new List<Point>();
+            List<Edge> edges = panel.ExternalEdges;
+            List<Face> faces = new List<Face>();
+            if (panel.Openings.Count()>0)
+            {
+                Engine.Reflection.Compute.RecordWarning("PanelToFEMesh does not support panel with opening");
+                return null;
+            }
+            if (panel.ExternalEdges.Count>4)
+            {
+                Engine.Reflection.Compute.RecordWarning("Panel contains more than 4 control points");
+                return null;
+            }
+            if (panel.ExternalEdges == null)
+                {
+                Engine.Reflection.Compute.RecordWarning("Checks identify no External edges");
+                return null;
+            }
+            if (panel == null)
+            {
+                Engine.Reflection.Compute.RecordWarning("Checks identify null panel");
+                return null;
+            }
```

Comment:
The `Panel` `null` check should come first, no point doing the rest of the checks otherwise, order of checks should be:
- `Panel == null`
- `Panel.ExternalEdges == null` and then check if any if any are `null` by looping over the `List<Edge>`
- If any of the `Curve` owned by the `Edge` is `null`
- Then if the `Curves` are Linear 
- Then if any of the points on the `Curve` is `null`

Then I would do all the circumstantial checks about number of edges etc.

Reply:
Would have a look into the nullcheck methods added by @IsakLarbornBH here. For example: `NullCheck(this Panel panel, string methodName = "Method")` that should be checking through all the geometrical aspects of the object. SImilar methods exists for the other elements as well.

Reply:
That method purposely doesn't do quite as thorough a check as it sounds like you're going for here, but it will check if the panel itself is null, has a non-zero amount of edges, and that each of those edges has a curve, so basically the first three steps Peter lists here.

Reply:
Thanks both, I will adapt some of the null checks into the code.